regparser/tree/xml_parser/interpretations.py

```diff
@@ -171,6 +190,24 @@ def process_inner_children(inner_stack, xml_node):
                     inner_stack.push_last((3 + par.depth, node))
                 else:
                     inner_stack.add(3 + par.depth, node)
+    elif nodes and manual_hierarchy_flag:
+        logging.warning('Using manual depth hierarchy.')
+        depths = PARAGRAPH_HIERARCHY[part][part_and_section]
+        if len(nodes) == len(depths):
+            for node, depth in zip(nodes, depths):
+                last = inner_stack.peek()
+                node.label = [l.replace('<E T="03">', '').replace('</E>', '')
+                                  for l in node.label]
+                if len(last) == 0:
+                    inner_stack.push_last((3 + depth, node))
```

Comment:
It came from the same place it came before; it looks like it's the calculation of what depth in the tree the section should begin at. I didn't invent this.